src/log-store/src/kafka/client_manager.rs

```diff
@@ -91,20 +92,18 @@ impl ClientManager {
         Ok(Self {
             config: config.clone(),
             client_factory: client,
-            client_pool: DashMap::new(),
+            client_pool: Mutex::new(HashMap::new()),
         })
     }
 
     /// Gets the client associated with the topic. If the client does not exist, a new one will
     /// be created and returned.
     pub(crate) async fn get_or_insert(&self, topic: &Topic) -> Result<Client> {
-        match self.client_pool.entry(topic.to_string()) {
-            DashMapEntry::Occupied(entry) => Ok(entry.get().clone()),
-            DashMapEntry::Vacant(entry) => {
-                let topic_client = self.try_create_client(topic).await?;
-                Ok(entry.insert(topic_client).clone())
-            }
+        let mut client_pool = self.client_pool.lock().await;
+        if let Entry::Vacant(entry) = client_pool.entry(topic.to_string()) {
+            entry.insert(self.try_create_client(topic).await?);
         }
+        Ok(client_pool[topic].clone())
```

Comment:
Use get to avoid cloning the topic each time.
```suggestion
        match client_pool.get(topic) {
            Some(client) => Ok(client.clone()),
            None => {
                let client = self.try_create_client(topic).await?;
                client_pool.insert(topic.to_string(), client.clone());
                Ok(client)
            }
        }
```